src/language-js/printer-estree.js

```diff
@@ -328,7 +329,13 @@ function printPathNoParens(path, options, print, args) {
             (node) =>
               node.type === "AwaitExpression" || node.type === "BlockStatement"
           );
-          if (parentAwaitOrBlock?.type !== "AwaitExpression") {
+          if (
+            parentAwaitOrBlock?.type !== "AwaitExpression" ||
+            !startsWithNoLookaheadToken(
```

Comment:
Are we checking node is the head of `AwaitExpression.argument` here?

This function seems not always checking the leftmost node, it skips on some nodes.

Reply:
This code is trying to avoid printing `await (await` on one line. Can you find a counterexample where it fails?

Reply:
No. Just a question, the result looks good to me.

Reply:
Even if it fails in some case, it won't break anything, just won't add forced indentation.